python/lsst/obs/base/ingest.py

```diff
@@ -570,8 +606,12 @@ def run(self, files, *, pool: Optional[Pool] = None, processes: int = 1, run: Op
                 runs.add(this_run)
             try:
                 with self.butler.transaction():
-                    refs.extend(self.ingestExposureDatasets(exposure, run=this_run))
+                    datasets_for_exposure = self.ingestExposureDatasets(exposure, run=this_run)
+                self._on_success(datasets_for_exposure)
```

Comment:
I think we need a try block around this. Otherwise if it fails it will immediately trigger an ingest failure even though it ingested just fine.

Reply:
Should I just put it in an `else` clause, so the callback _could_ raise, but we wouldn't incorrectly classify it as an ingest failure?  (And maybe document that if it does raise, it'll bring down the whole ingest run.)

Reply:
Putting it in the else seems like the right thing to do (and documenting that it will stop ingest).